pkg/machines/components/storagePools/storageVolumeDelete.jsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+    Button,
+    OverlayTrigger,
+    Tooltip,
+} from 'patternfly-react';
+
+import { deleteStorageVolume } from '../../libvirt-dbus.js';
+import cockpit from 'cockpit';
+
+import './storageVolumeDelete.css';
+
+const _ = cockpit.gettext;
+
+export class StorageVolumeDelete extends React.Component {
+    constructor(props) {
+        super(props);
+        this.storageVolumeListDelete = this.storageVolumeListDelete.bind(this);
+    }
+
+    storageVolumeListDelete() {
+        const { volumes, storagePool, resetSelection } = this.props;
+
+        Promise.all(volumes.map(volume =>
+            deleteStorageVolume({
+                poolName: storagePool.name,
+                connectionName: storagePool.connectionName,
+                volumePath: [storagePool.path, volume].join('/'),
+            })
+        ))
+                .catch(exc => {
+                    this.props.deleteErrorHandler(_("Storage Volumes could not be deleted"), exc.message);
+                })
+                .then(() => {
+                    resetSelection();
+                });
+    }
+
+    render() {
+        const { volumes, isVolumeUsed } = this.props;
+        const volCount = volumes.length;
+        const anyVolumeUsed = volumes.some(volume => isVolumeUsed[volume].length != 0);
+
+        if (volCount == 0)
+            return null;
+
+        const deleteBtn = (
+            <Button className='storage-volumes-actions' id='storage-volumes-delete'
+                    bsStyle='danger' onClick={this.storageVolumeListDelete}
+                    disabled={ anyVolumeUsed }>
+                {`${_("Delete")} ${volCount} ${volCount > 1 ? 'volumes' : 'volume'}`}
```

Comment:
This used to bring up a Modal for confirmation, but not any more. That was done on purpose? (FTR, I think it's fine -- it's only in the expander details, so you can't accidentally delete your storage volume by a mis-click into the original list, right?)

Reply:
It was removed by mistake. Just pushed from another branch that didn't have this change :/ That's why I should not ignore PRs for such a long time.

Reply:
So, I remember that we/I decided not not have a delete modal open on deletion, since we are allowing deletion only of volumes that are not used.  So it's not such a great deal to delete an unused volume.